Source/components/Image.js

```diff
@@ -0,0 +1,52 @@
+import { DefinedPropertiesKey } from '../WrappedObject'
+import { Layer } from './Layer'
+import { Rectangle } from '../Rectangle'
+import { Types } from '../enums'
+import { Factory } from '../Factory'
+
+/**
+ * Represents an image layer.
+ */
+export class Image extends Layer {
+  /**
+   * Make a new image layer object.
+   *
+   * @param {MSBitmapLayer} layer The underlying model object from Sketch.
+   */
+  constructor(layer = {}) {
+    if (!layer.sketchObject) {
+      // eslint-disable-next-line no-param-reassign
+      layer.sketchObject = Factory.createNative(Image)
+        .alloc()
+        .initWithFrame(new Rectangle(0, 0, 100, 100).asCGRect())
+    }
+
+    super(layer)
+  }
+}
+
+Image.type = Types.Image
+Image[DefinedPropertiesKey] = { ...Layer[DefinedPropertiesKey] }
+Factory.registerClass(Image, MSBitmapLayer)
+
+Image.define('image', {
+  get() {
+    // TODO: return something nice here
+    return this._object.image
+  },
+  set(image) {
+    let nsImage
+    if (typeof image === 'string') {
+      nsImage = NSImage.alloc().initByReferencingFile(image)
+    } else if (image.class && String(image.class()) === 'NSImage') {
+      nsImage = image
+    } else if (image.class && String(image.class()) === 'NSURL') {
+      nsImage = NSImage.alloc().initWithContentsOfURL_(image)
```

Comment:
that was the existing code. But it's a good point and we should be more careful.

Maybe throw an error instead of crashing? That's would be another PR tho

Reply:
The crash comes from Objective-C, I believe, not us. I'll open a new PR